IsyFact/isyfact-standards
# IFS-2400 Security- and Authentifizierungsmanager interface

- [ ] depends on https://github.com/IsyFact/isyfact-standards/pull/40

related commits:
* test: IFS-2400 cover both flows with assertions
* feat: IFS-2400 add header for bhknz
* feat: IFS-2400 add property for standard OU
* feat: IFS-2400 enable caching for client credentials authentication by client registration
* feat: IFS-2400 split autoconfiguration for conditional bean creation
* feat: IFS-2400 bundle common functionality of Isy..AuthenticationProviders in abstract superclass
* feat: IFS-2400 rename AutoConfiguration
* test: IFS-2400 check authorizedClient caching is used
* test: IFS-2400 test autoconfiguration conditions
* chore: IFS-2400 update CHANGELOG.md
* style: IFS-2400 remove checkstyle warnings

    * add javadoc
    * add package-info

* docs: IFS-2400 describe usage of the Security and Authentifizierungsmanager interface
* feat: IFS-2400 move IsyOAuth2Authentifizierungsmanager to core package with @NonNullApi
* feat: IFS-2400 interface method signature changed to sort the parameters in logical order

    * spring -> isyfact specific -> optional

* feat: IFS-2400 rename default implementations with IsyOAuth2* prefix

    * indicate class is isyfact specific and OAuth2 related

* chore: IFS-2400 fix javadoc typo
* test: IFS-2400 move AutoConfiguration specific Testcase into IsySecurityAutoConfigurationTest
* feat: IFS-2400 add support for @AutoConfiguration, improve javadoc, fix typos
* refactor: IFS-2400 move oauth client related classes to the oauth.client package
* refactor: IFS-2400 shorten names of provider classes for better readability, add minor improvements
* refactor: IFS-2400 move Security and Berechtigungsmanager interfaces to core package
* feat: IFS-2400 refactor oauth classes for better testability
* test: IFS-2400 add more tests, refactor existing tests
* fest: IFS-2400 fix NPE in RolePrivilegesMapper
* build: IFS-2400 cleanup maven dependencies (see IFS-2481)
